src/Sarif.Multitool.FunctionalTests/TestData/UrisMustBeValid/Invalid.sarif

```diff
@@ -0,0 +1,61 @@
+{
+  "$schema": "ht%tp://json.schemastore.org/sarif-2.0.0",
+  "version": "2.0.0",
+  "runs": [
+    {
+      "tool": {
+        "name": "CodeScanner",
+        "downloadUri": "ht%tp://www.example.com/tools/codescanner/download.html"
+      },
+      "originalUriBaseIds": {
+        "SRCROOT": {
+          "uri": "fi%le:///c:/Code/sarif-sdk/src"
+        }
+      },
+      "results": [
+        {
+          "ruleId": "TST0001",
+          "level": "error",
+          "analysisTarget": {
+            "uri": "fi&le:///c:/src/file.c"
+          },
+          "workItemUris": [
+            "ht&tp://example.com/my-project/issues/42"
+          ]
+        }
+      ],
+      "files": {
+        "fi%le:///c:/src/file.c": {
+          "mimeType": "text/x-c"
+        }
+      },
+      "versionControlProvenance": [
+        {
+          "repositoryUri": "ht%tps://example.com/my-project"
+        }
+      ],
+      "resources": {
+        "rules": {
+          "TST0001": {
+            "id": "TST0001",
+            "helpUri": "ht%tp://www.example.com/rules/tst0001.html"
+          }
+        }
+      },
+      "properties": {
```

Comment:
>properties [](start = 7, length = 10)

Ok. I still have concerns about the fragility of this mechanism (for example, the need to change the index). The implied path changes seem trivial to update, however. And I get that you are providing data that points to to primitive values. I have had issues with this sort of approach in the past where I have attempted to constrain the reporting to a single check. Sometimes it is impossible (and perhaps not even desirable) to limit things in this way. In which chase, it is helpful to be able to comprehensively describe the rule ids that are fired for certain patterns.